app/screens/login/Login.js

```diff
@@ -0,0 +1,93 @@
+// React
+import React from 'react';
+// React Native
+import { Alert, AsyncStorage, Image, KeyboardAvoidingView, View } from 'react-native';
+// React Native Elements
+import { Header, Input } from 'react-native-elements';
+// CSS
+import globalStyles from '../../theme/styles';
+// Icons
+import Feather from "react-native-vector-icons/Feather";
+import AntDesign from "react-native-vector-icons/AntDesign";
+// Services
+import firebaseAuth from '../../services/firebaseAuth/firebaseAuth';
+// Helpers
+import { fieldChange, showHide, navigateTo } from '../../helpers/inputHelper/inputHelper';
+// Components
+import Logos from '../../components/Logos/Logos';
+import NavigateToButton from '../../components/NavigateToButton/NavigateToButton';
+
+class Login extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userName: '',
+      password: '',
+      showPassword: true
+    };
+    this.fireAuth = new firebaseAuth();
+  }
+
+  fieldChanged = (text, fieldName) => {
+    this.setState(fieldChange(text, fieldName));
+  }
+
+  showHidePassword = (field) => {
+    this.setState({[field]: showHide(field, this.state)});
+  }
+
+  loginAsync = async (logedUser) => {
+    this.setState({userName: '', password: '', showPassword: true});
+    await AsyncStorage.setItem('uid', `${logedUser}`);
+    this.props.navigation.navigate('AuthLoading');
+  };
+
+  loginUser = async () => {
+    const currentState = {...this.state};
+    const { userName, password } = currentState;
+    if ((userName) && (password)) {
+      const signedUser = await this.fireAuth.loginUser(userName, password);
+      if (signedUser.uid) {
+        this.loginAsync(signedUser.uid);
+      } else {
+        Alert.alert(`${signedUser}`);
+      }
+    } else {
+      Alert.alert('there are empty fields');
+    }
+  }
+
+  render() {
```

Comment:
This render has a lot of logic, You can refarctor this, break down the code in functions and would be more readable and clean.